File: 백준/Silver/15651. N과 M （3）/N과 M （3）.js
const [N, M] = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split(" ")
  .map(Number);

const sequence = [];
let result = "";

const dfs = (count) => {
  if (count === M) {
    result += sequence.join(" ") + "\n";
    return;
  }

  for (let i = 0; i < N; i++) {
    sequence.push(i + 1);
    dfs(count + 1);
    sequence.pop();
  }
};

dfs(0);

console.log(result);

File: 백준/Silver/15651. N과 M （3）/N과 M （3）.py
N, M = map(int, input().split())

# 1부터 N까지 자연수에서 중복없이 M개 고른 수열
# 중복가능


stack = []
visited = [0] * (N+1)


def dfs(depth):
    if depth == M:
        print(*stack)
        return

    for i in range(1, N+1):
        stack.append(i)
        dfs(depth+1)
        stack.pop()


dfs(0)
